File: Ingsoft/RistoranteWeb/Models/Richiesta/Richiesta.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace RistoranteWeb.Models.Richiesta
{
    public abstract class Richiesta
    {
        private string _nominativo;
        private string _IDRichiesta;
        private DateTime _dataOra;

        public string Nominativo { get => _nominativo; set => _nominativo = value; }
        public string IDRichiesta { get => _IDRichiesta; set => _IDRichiesta = value; }
        public DateTime DataOra { get => _dataOra; set => _dataOra = value; }

        protected Richiesta(string nominativo, DateTime dataOra)
        {
            _nominativo = nominativo;
            _dataOra = dataOra;
        }
    }
}

File: FortunaWeb/fortuna/src/main/java/it/unibo/ingsoft/fortuna/model/richiesta/Richiesta.java
package it.unibo.ingsoft.fortuna.model.richiesta;

import java.time.LocalDateTime;

import org.springframework.validation.ObjectError;

public abstract class Richiesta {
    private String nominativo;
    private String idRichiesta;
    private LocalDateTime dataOra;

    protected Richiesta() {}

    protected Richiesta(String nominativo, LocalDateTime dataOra)
    {
        this.nominativo = nominativo;
        this.dataOra = dataOra;
    }


    public String getNominativo() {
        return this.nominativo;
    }

    public void setNominativo(String nominativo) {
        this.nominativo = nominativo;
    }

    public String getIdRichiesta() {
        return this.idRichiesta;
    }

    public void setIdRichiesta(String idRichiesta) {
        this.idRichiesta = idRichiesta;
    }

    public LocalDateTime getDataOra() {
        return this.dataOra;
    }

    public void setDataOra(LocalDateTime dataOra) {
        this.dataOra = dataOra;
    }


    @Override
    public boolean equals(Object o) {
        if (o == this)
            return true;
        if (!(o instanceof Richiesta)) {
            return false;
        }
        Richiesta richiesta = (Richiesta) o;
        return this.getIdRichiesta().equals(richiesta.getIdRichiesta());
    }
}
